lib/ansible/config/manager.py

```diff
@@ -399,8 +399,14 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
         if config in defs:
 
             # direct setting via plugin arguments, can set to None so we bypass rest of processing/defaults
-            if direct and config in direct:
-                value = direct[config]
+            direct_aliases = []
+            if direct:
+                direct_aliases = [direct[alias] for alias in defs[config].get('aliases', []) if alias in direct]
+            if direct and (config in direct or direct_aliases):
+                if config in direct:
+                    value = direct[config]
+                else:
+                    value = direct_aliases[0]
```

Comment:
won't you get an error when no aliases?

Reply:
I'm checking `direct_aliases` is truthy or `config in direct` above, so this shouldn't error.

Reply:
I could clarify with an elif if it helps readability, i.e.
```
            if direct and config in direct:
                value = direct[config]
            elif direct and direct_aliases:
                value = direct_aliases[0]
```